generators/client/templates/vue/src/main/webapp/app/components/admin/health/health.component.ts.ejs

```diff
@@ -1,57 +1,65 @@
+import HealthService from './health.service';
 import <%=jhiPrefixCapitalized%>HealthModal from './health-modal.vue';
-import HealthService from './HealthService.vue';
+import { Component } from 'vue-property-decorator';
+import { mixins } from 'vue-class-component';
 
-const <%=jhiPrefixCapitalized%>HealthComponent = {
-  name: '<%=jhiPrefixCapitalized%>HealthComponent',
-  mixins: [HealthService],
+const beforeRouteEnter = (to, from, next) => {
```

Comment:
For such a case, when the `beforeRouteEnter` does not use any of `to`, `from` & `next`, I'd rather create put this code inside `mounted`: 

```typescript
public mounted(): void {
  this.refresh();
}
```

Reply:
I tried with `mounted`, the tests did not pass because of a Promise mistake

Reply:
It's because you need to mock axios in the unit test:

```typescript
beforeEach(() => {
  mockedAxios.get.mockReturnValue(Promise.resolve({}));
  wrapper = shallowMount<HealthClass>(Health, { store, i18n, localVue });
  health = wrapper.vm;
});
```